torchlars/lars.py

```diff
@@ -50,26 +50,24 @@ def __init__(self, optimizer, eps=1e-8, trust_coef=0.001):
         self.trust_coef = trust_coef
         self.adaptive_lr = torch.ones([])
 
-    def __copy__(self):
-        cls = self.__class__
-        result = cls.__new__(cls)
-        result.__dict__.update(self.__dict__)
-        return result
-
-    def __deepcopy__(self, memo):
-        import copy
-        cls = self.__class__
-        result = cls.__new__(cls)
-        memo[id(self)] = result
-        for k, v in self.__dict__.items():
-            setattr(result, k, copy.deepcopy(v, memo))
-        return result
-
     def __getstate__(self):
-        return self.optim.__getstate__()
+        core_optim_dict = self.optim.__getstate__()
+        lars_dict = {}
+        lars_dict['class_name'] = self.optim.__class__
+        lars_dict['eps'] = self.eps
+        lars_dict['trust_coef'] = self.trust_coef
+        lars_dict['adaptive_lr'] = self.adaptive_lr
+        return (core_optim_dict, lars_dict)
 
     def __setstate__(self, state):
-        self.optim.__setstate__(state)
+        core_optim_dict, lars_dict = state
```

Comment:
If `__getstate__()` returns `self.optim`, this line can be rewritten as:

```suggestion
        self.optim, lars_dict = state
```